torch/_dynamo/variables/dicts.py

```diff
@@ -413,22 +402,12 @@ def call_method(
         if name == "add" and args and self.mutable_local:
             assert not kwargs
             item = args[0]
-            result = SetVariable(
-                self._add(item),
-                mutable_local=self.mutable_local,
-            )
-            tx.replace_all(self, result)
+            self._add(item)
             return ConstantVariable.create(None)
         elif name == "pop" and self.mutable_local:
             assert not kwargs
             assert not args
-            items = list(self.items)
-            result = items.pop()
-            tx.replace_all(
-                self,
-                SetVariable(items),
-            )
-            return result
+            return self.items.pop()
```

Comment:
Missing call to mutation?

Reply:
Hmm I'll add this, but right now I don't think sets track mutation at all since we don't replay mutations for them at the end so it's in essence a no-op. 

I can add mutation tracking for sets in a follow up PR

Reply:
Don't we need to assert there aren't side effects escaping the scope?